HeterogeneousCore/SonicCore/interface/SonicClientPseudoAsync.h

```diff
@@ -0,0 +1,76 @@
+#ifndef HeterogeneousCore_SonicCore_SonicClientPseudoAsync
+#define HeterogeneousCore_SonicCore_SonicClientPseudoAsync
+
+#include "FWCore/Concurrency/interface/WaitingTaskWithArenaHolder.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientBase.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientTypes.h"
+
+#include <memory>
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+#include <atomic>
+#include <exception>
+
+//pretend to be async + non-blocking by waiting for blocking calls to return in separate std::thread
+template <typename InputT, typename OutputT = InputT>
+class SonicClientPseudoAsync : public SonicClientBase, public SonicClientTypes<InputT, OutputT> {
+public:
+  //constructor
+  SonicClientPseudoAsync() : SonicClientBase(), SonicClientTypes<InputT, OutputT>(), hasCall_(false), stop_(false) {
+    thread_ = std::make_unique<std::thread>([this]() { waitForNext(); });
```

Comment:
A thread per module x stream is fine for now, but I created an issue (#/29188) for a more general mechanism in case more use cases would surface.